File: java-poc/src/main/java/io/craigmiller160/querydslpoc/java/service/QueryDSLService.java
package io.craigmiller160.querydslpoc.java.service;

import com.querydsl.jpa.JPQLQueryFactory;
import io.craigmiller160.querydslpoc.java.data.entity.Employee;
import io.craigmiller160.querydslpoc.java.data.entity.QEmployee;
import java.util.Optional;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;

@Service
@RequiredArgsConstructor
public class QueryDSLService {
  private final JPQLQueryFactory queryFactory;

  public Optional<Employee> getOneEmployee() {
    final Employee nullableResult =
        queryFactory
            .query()
            .from(QEmployee.employee)
            .limit(1)
            .select(QEmployee.employee)
            .fetchFirst();
    return Optional.ofNullable(nullableResult);
  }
}

File: kotlin-poc/src/main/kotlin/io/craigmiller160/querydslpoc/kotlin/service/QueryDSLService.kt
package io.craigmiller160.querydslpoc.kotlin.service

import com.querydsl.jpa.impl.JPAQueryFactory
import io.craigmiller160.querydslpoc.kotlin.data.entity.Employee
import io.craigmiller160.querydslpoc.kotlin.data.entity.QEmployee
import org.springframework.stereotype.Service

@Service
class QueryDSLService(private val queryFactory: JPAQueryFactory) {
  fun getOneEmployee(): Employee? =
    queryFactory.query().from(QEmployee.employee).limit(1).select(QEmployee.employee).fetchFirst()
}
